### PR TITLE
Github OAuth integration

### PR DESCRIPTION
**Pull request by gsmithun4**
Resolves https://github.com/ToolJet/ToolJet/issues/1828

### CODE REVIEW

File: server/ee/services/oauth/oauth.service.ts [hunk 1]
```diff
@@ -14,20 +16,41 @@ export class OauthService {
...
+      this.configService.get<string>('SSO_DISABLE_SIGNUPS') &&
+      this.configService.get<string>('SSO_DISABLE_SIGNUPS') === 'true';
+    this.restrictedDomain = this.configService.get<string>('RESTRICTED_DOMAIN');
+  }
+
+  private readonly ssoSignUpDisabled: boolean;
+  private readonly restrictedDomain: string;
 
-  async #findOrCreateUser({ userSSOId, firstName, lastName, email }): Promise<User> {
+  #isValidDoamin(domain: string): boolean {
```
Comment (akshaysasidrn) [inline server/ee/services/oauth/oauth.service.ts:32]: domain*

**Review by akshaysasidrn — APPROVED** [review]
Nice to make use of interfaces. Easier to reason about functions.

**Comment by Navaneeth-pk** [comment]
After signing in with GitHub, a toast appears "Invalid email or password". Can you please check? @gsmithun4

**Comment by Navaneeth-pk** [comment]
If signed in using GitHub, the ( new ) user doesn't have app creation permissions. @gsmithun4

**Comment by gsmithun4** [comment]
> After signing in with GitHub, a toast appears "Invalid email or password". Can you please check? @gsmithun4

@Navaneeth-pk Fixed in `Supress initial error thrown by google` commit. let me check if the changes are deployed

**Review by akshaysasidrn — APPROVED** [review]

**Pull request merged**